### PR TITLE
Add devices learning area

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This PR is about adding a brand new "Devices" learning area.
Web developers will learn to communicate with hardware devices over Bluetooth, USB, NFC, Serial, and HID from a website.
This area will get more advanced articles in the future such as "building devices designed for WebUSB".

~~My goal is to move Bluetooth and USB articles from dgc to web.dev.
In the mean time, I'm adding redirects for those two pages so that web.dev/bluetooth and web.dev/usb are valid URLs for CDS session. Sadly, it looks like redirects are 301 (moved permanently) which is not what I want ;(~~

~~Note that I'm missing a nice cover asset for this section (it is marked as **TODO**).~~

Live preview:
- https://deploy-preview-3980--web-dev-staging.netlify.app/learn/
- https://deploy-preview-3980--web-dev-staging.netlify.app/devices/

### CODE REVIEW

**Comment by beaufortfrancois** [comment]
@kaycebasques What do you think?

**Comment by beaufortfrancois** [comment]
Looking at what has been done for `Scale on web", I wonder if I should simply tag devices posts and add a redirect from  https://web.dev/devices to https://web.dev/tags/devices

**Comment by jpmedley** [comment]
I think it would be worth doing whether you eventually create a devices/
section or not. There are variations in how tags could have been
implemented. They work the way they do on our site for uses cases just like
this.
Joe Medley | Technical Writer, Chrome DevRel | jmedley@google.com |
 816-678-7195
*If an API's not documented it doesn't exist.*

**Comment by beaufortfrancois** [comment]
Following https://github.com/GoogleChrome/web.dev/pull/3980#issuecomment-702718817, I've created https://github.com/GoogleChrome/web.dev/pull/4001 to add a new "devices" tag in the meantime.

**Comment by kaycebasques** [comment]
I hesitated over using "devices" (e.g. `https://web.dev/devices`) but I can't think of a better phrase. We could always change it later if something else comes up.

@jimoong can you create a graphic (i.e. an illustration for https://web.dev/learn/) for this new collection (i.e. "learning path")? How long will it take? Long story short, it's a collection of APIs for controlling hardware devices (serial, USB, etc.). LMK if you need more direction.

**Review by kaycebasques — COMMENTED** [review]
@beaufortfrancois I generally support the idea.

While we are here, here are a couple of ideas:

* Take a look at [Divio's Documentation System](https://documentation.divio.com/introduction/#the-secret) which outlines the four common types of documentation (tutorials, how-to guides, references, explanations). Are there any tutorials, how-to guides, or explanations (also known as overviews) that could tie this collection together?
* I have a tool that makes it easier to migrate content to web.dev. Let's just migrate the old developers.google.com/web/updates posts now?

P.S. you've got a merge conflict in the redirects file.

**Comment by jpmedley** [comment]
@beaufortfrancois FYI for when you're done. This is code, so I don't have the authority to merge.

**Comment by jimoong** [comment]
12 devices

Hey, sorry for belated update. Here is a version of illustration that I tried today for the device section. Would something like this work for it? Feel free to let me know what you think.

**Comment by beaufortfrancois** [comment]
I love it!
Can you share the svg file?

**Comment by kaycebasques** [comment]
@beaufortfrancois I'm testing out my migration tool on your target pages (I realized it was too early to ask you to do it because I've barely used it on webfu) and I'll upload drafts myself (if the conversion works well)

**Comment by jimoong** [comment]
Awesome! Here is the SVG file.



*Jiwoong Lee**Interaction Designer, Google*

**Comment by beaufortfrancois** [comment]
The svg file is missing. Can you share it with me through email?

**Comment by beaufortfrancois** [comment]
@kaycebasques @robdodson With the new illustration, does it look good to you?

File: src/images/collections/devices.svg [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <path fill-rule="evenodd" clip-rule="evenodd" d="M197.93 110A8 8 0 11191 98h18a8 8 0 11-6.93 12h-4.14z"/>
+  </mask>
+  <path fill-rule="evenodd" clip-rule="evenodd" d="M197.93 110A8 8 0 11191 98h18a8 8 0 11-6.93 12h-4.14z" fill="#fff"/>
+  <path d="M197.93 110v-1h-.577l-.289.499.866.501zm4.14 0l.866-.501-.289-.499h-.577v1zM191 115a8.996 8.996 0 007.795-4.499l-1.731-1.002A6.997 6.997 0 01191 113v2zm-9-9a9 9 0 009 9v-2a7 7 0 01-7-7h-2zm9-9a9 9 0 00-9 9h2a7 7 0 017-7v-2zm0 2h18v-2h-18v2zm27 7a9 9 0 00-9-9v2a7 7 0 017 7h2zm-9 9a9 9 0 009-9h-2a7 7 0 01-7 7v2zm-7.795-4.499A8.996 8.996 0 00209 115v-2a6.997 6.997 0 01-6.064-3.501l-1.731 1.002zm.865-1.501h-4.14v2h4.14v-2z" fill="#212121" mask="url(#a)"/>
+  <circle cx="209" cy="103" r="1.5" fill="#212121"/>
+  <circle cx="209" cy="109" r="1.5" fill="#212121"/>
+  <circle cx="212" cy="106" r="1.5" fill="#212121"/>
+  <circle cx="206" cy="106" r="1.5" fill="#212121"/>
+  <path d="M191 102.5v7m-3.5-3.5h7" stroke="#212121"/>
+</svg>
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/images/collections/devices.svg:44]: Asset was already done with this tool ;)

**Comment by beaufortfrancois** [comment]
https://github.com/GoogleChrome/web.dev/pull/4090 is tracking moving the Web Bluetooth article from WebFu to web.dev

**Comment by beaufortfrancois** [comment]
@kaycebasques @jpmedley FYI I've added a new "CDS 2020" label to the devices related PRs.

**Comment by beaufortfrancois** [comment]
https://github.com/GoogleChrome/web.dev/pull/4095 is tracking moving the WebUSB article from WebFu to web.dev

**Comment by beaufortfrancois** [comment]
> @beaufortfrancois please try out this markdown that my migration tool generated: https://drive.google.com/file/d/1m7AQZIfpfnA4ukNyfVv-kUd3DrbdS60f/view?usp=sharing
> 
> LMK if anything got borked. Things I'm aware of:
> 
> * Still references the old image paths
> * The code samples that use syntax highlighting on the Bluetooth page didn't migrate correctly

For info, I ended up migrating it manually without the tool as all links were missing.

File: src/images/collections/devices.svg (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaycebasques) [reply src/images/collections/devices.svg:44]: Ah, OK. Usually when I use that tool the output is minified one-line markup which is why I thought it wasn't.

File: src/site/content/_redirects.yaml [hunk 2]
```diff
@@ -230,6 +230,13 @@ redirects:
 - from: /native-lazy-loading
   to: /browser-level-image-lazy-loading
 
+# Temporary redirects for devices API
+- from: /bluetooth
+  to: https://developers.google.com/web/updates/2015/07/interact-with-ble-devices-on-the-web
+
+- from: /usb
+  to: https://developers.google.com/web/updates/2016/03/access-usb-devices-on-the-web
+
```
Comment (kaycebasques) [inline src/site/content/_redirects.yaml:239]:
```suggestion
```
No longer needed

**Review by kaycebasques — APPROVED** [review]

**Comment by kaycebasques** [comment]
@beaufortfrancois thanks for letting me know that my migration tool was goofing up links (I made a last-minute configuration change and didn't notice all the missing links… it's fixed now). If you ever find yourself migrating content again ping me if you think the tool will be helpful.

**Comment by kaycebasques** [comment]
@robdodson or @samthor PTAL at the JS changes

File: src/site/_data/paths/devices.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  "overview": "Learn to communicate with hardware devices over Bluetooth, USB, NFC, Serial, and HID from a website.",
+  "topics": [
+    {
+      "title": "API Guides",
+      "pathItems": [
+        "nfc",
+        "serial",
+        "hid",
+        "usb",
+        "bluetooth"
```
Comment (robdodson) [inline src/site/_data/paths/devices.json:16]: these two guides don't seem to exist?

**Review by robdodson — APPROVED** [review]
js changes LGTM

File: src/site/_data/paths/devices.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "slug": "devices",
+  "cover": "/images/collections/devices.svg",
+  "title": "Devices",
+  "updated": "October 15, 2020",
+  "description": "Connecting hardware devices to the web",
+  "overview": "Learn to communicate with hardware devices over Bluetooth, USB, NFC, Serial, and HID from a website.",
+  "topics": [
+    {
+      "title": "API Guides",
```
Comment (kaycebasques) [inline src/site/_data/paths/devices.json:10]:
```suggestion
      "title": "API guides",
```

**Pull request merged**